### PR TITLE
Feature/initial version

### PR DESCRIPTION
**Pull request by louisfelix**
Adding first metrics (resource_health_availability_up and optional azure_tag_info)

### CODE REVIEW

File: resources.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+	if err != nil {
+		return nil, err
+	}
+
+	// Filtering by tag is done manually, as Azure does not support
+	// to filter both by resource type and by tag name/value
+	var filteredList []resources.GenericResource
+	for _, resource := range *resList {
+		include := true
+		if resourceTags != nil {
```
Comment (rolandvarga) [inline resources.go]: Should we break the loop if `resourceTags == nil`? I can foresee a scenario where that happens and `include` will stay `true`, hence appending all resources to the list.

File: resource_health_collector.go [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+			}
+		}
+	}
+}
+
+// CollectAvailabilityUp converts Resource Health Availability status as an UP metric
+func (c *ResourceHealthCollector) CollectAvailabilityUp(ch chan<- prometheus.Metric, as *resourcehealth.AvailabilityStatus,
+	resource *resources.GenericResource) {
+
+	// Only the `Unavailable` status can be used with confidence to considere availability "down"
```
Comment (rolandvarga) [inline resource_health_collector.go]: fix typo `consider`

File: main.go [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+	err = yaml.Unmarshal(configData, &config)
+	if err != nil {
+		return config, err
+	}
+
+	log.Info("Config loaded")
+	return config, nil
+}
+
+func fileExists(filename string) bool {
```
Comment (rolandvarga) [inline main.go]: This is just an observation, but I believe recurring methods as this should be included in a generic 'utils' package.

Comment (louisfelix) [reply main.go]: Yes, you are totally right. We will consider this soon I think.

File: resource_health_collector.go (hunk 2 above)
Comment (louisfelix) [reply resource_health_collector.go]: done

File: resources.go (hunk 1 above)
Comment (louisfelix) [reply resources.go]: That was indeed the intended behaviour; the `resource_tags` config was optional, and all resources of the given type got selected if no tag filtering is done. But that may be counter-intuitive after all. I will change that.

Comment (louisfelix) [reply resources.go]: Fixed

**Pull request merged**